File: Bookstore_Management_System/inventoryclient/src/main/java/inventoryclient/LowStockBooksFetcher.java
package inventoryclient;

import org.json.JSONArray;
import org.json.JSONObject;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class LowStockBooksFetcher {

    private static final String SERVICE_URL = "http://localhost/inventorymanagement/public/api/";

    public static void main(String[] args) {
        try {
            getLowStockBooks();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    private static void getLowStockBooks() throws Exception {
        URL url = new URL(SERVICE_URL + "getLowStockBooks.php");
        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
        conn.setRequestMethod("GET");

        BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
        String inputLine;
        StringBuilder response = new StringBuilder();

        while ((inputLine = in.readLine()) != null) {
            response.append(inputLine);
        }
        in.close();

        JSONArray lowStockBooks = new JSONArray(response.toString());
        System.out.println("Books with low stock:");
        for (int i = 0; i < lowStockBooks.length(); i++) {
            JSONObject book = lowStockBooks.getJSONObject(i);
            System.out.println("Book ID: " + book.getInt("id"));
            System.out.println("Title: " + book.getString("title"));
            System.out.println("Stock: " + book.getInt("stock_quantity"));
        }
    }
}

File: Bookstore_Management_System/inventoryclient/src/main/python/inventoryclient/LowStockBooksFetcher.py
import requests

SERVICE_URL = "http://localhost/Bookstore_Management_System\BookStore\public\api\getLowStockBooks.php"

def get_low_stock_books():
    response = requests.get(SERVICE_URL)
    if response.status_code == 200:
        low_stock_books = response.json()
        print("Books with low stock:")
        for book in low_stock_books:
            print(f"Book ID: {book['id']}")
            print(f"Title: {book['title']}")
            print(f"Stock: {book['stock_quantity']}")
    else:
        print("Failed to fetch low stock books.")

if __name__ == "__main__":
    get_low_stock_books()
